com.unity.render-pipelines.high-definition/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl

```diff
@@ -1,484 +1,243 @@
-        return debugColors;
-    }
-    else if (_DebugProbeVolumeMode == PROBEVOLUMEDEBUGMODE_VISUALIZE_VALIDITY)
-    {
-        float validity = coefficients.data[0].x;
-        return lerp(float3(1, 0, 0), float3(0, 1, 0), validity);
-    }
-    else
-#endif
-    {
-        float3 sampleOutgoingRadiance = coefficients.data[0].rgb;
-        return sampleOutgoingRadiance;
+        return false;
     }
+
+    // unpack pool idx
+    // size is encoded in the upper 4 bits
+    uint   subdiv = (packed_pool_idx >> 28) & 15;
+    float  cellSize = pow(3.0, subdiv);
+    uint   flattened_pool_idx = packed_pool_idx & ((1 << 28) - 1);
+    uint3  pool_idx;
+    pool_idx.z = flattened_pool_idx / (apvConst.poolDim.x * apvConst.poolDim.y);
+    flattened_pool_idx -= pool_idx.z * (apvConst.poolDim.x * apvConst.poolDim.y);
+    pool_idx.y = flattened_pool_idx / apvConst.poolDim.x;
+    pool_idx.x = flattened_pool_idx - (pool_idx.y * apvConst.poolDim.x);
+    uvw = ((float3) pool_idx + 0.5) / (float3) apvConst.poolDim;
+
+    // calculate uv offset and scale
+    float3 offset = frac(posRS / (float)cellSize);  // [0;1] in brick space
+    //offset    = clamp( offset, 0.25, 0.75 );      // [0.25;0.75] in brick space (is this actually necessary?)
+    offset *= 3.0 / (float3) apvConst.poolDim;      // convert brick footprint to texels footprint in pool texel space
+    uvw += offset;                                  // add the final offset
+
+    return true;
 }
 
-float3 ProbeVolumeEvaluateSphericalHarmonicsL1(float3 normalWS, ProbeVolumeSphericalHarmonicsL1 coefficients)
+void EvaluateAdaptiveProbeVolume(in float3 posWS, in float3 normalWS, in float3 backNormalWS, in APVResources apvRes,
+    out float3 bakeDiffuseLighting, out float3 backBakeDiffuseLighting)
 {
+    bakeDiffuseLighting = float3(0.0, 0.0, 0.0);
+    backBakeDiffuseLighting = float3(0.0, 0.0, 0.0);
 
-#ifdef DEBUG_DISPLAY
-    if (_DebugProbeVolumeMode == PROBEVOLUMEDEBUGMODE_VISUALIZE_DEBUG_COLORS)
-    {
-        float3 debugColors = coefficients.data[0].rgb;
-        return debugColors;
-    }
-    else if (_DebugProbeVolumeMode == PROBEVOLUMEDEBUGMODE_VISUALIZE_VALIDITY)
+    float3 pool_uvw;
+    if (TryToGetPoolUVW(apvRes, posWS, normalWS, pool_uvw))
     {
-        float validity = coefficients.data[0].x;
-        return lerp(float3(1, 0, 0), float3(0, 1, 0), validity);
+        float3 L0 = EvaluateAPVL0(apvRes, pool_uvw);
+        EvaluateAPVL1(apvRes, L0, normalWS, backNormalWS, pool_uvw, bakeDiffuseLighting, backBakeDiffuseLighting);
+
+        bakeDiffuseLighting += L0;
+        backBakeDiffuseLighting += L0;
     }
     else
-#endif
     {
-        float3 sampleOutgoingRadiance = SHEvalLinearL0L1(normalWS, coefficients.data[0], coefficients.data[1], coefficients.data[2]);
-        return sampleOutgoingRadiance;
+        // no valid brick, fallback to ambient probe
+        bakeDiffuseLighting = EvaluateAmbientProbe(normalWS);
```

Comment:
See my comment above, same code is use for both front and back lighting in builtinGIUtilitiers.hlsl